design/one-pager-release-process.md

```diff
@@ -0,0 +1,172 @@
+1. The release process itself has a lot of manual steps that are time consuming, error prone, and insecure.
+   The cost of performing a release is high and inhibits faster releases from happening.
+
+## Proposed Improvements
+
+In this section, we will cover the set of proposed changes to improve our release engineering and process.
+
+## Summary
+
+For convenience, a summary of the proposal is included below:
+
+* Increase the frequency of releases to a consistent **monthly** schedule
+* Follow [semantic versioning](https://semver.org) so that each monthly release is a minor release (e.g. `v0.4.0`)
+* Run patch releases (e.g., `v0.4.1`) as needed to address issues with quality and stability in the previous release
+* Add pipelines to Jenkins that leverage the [build submodule](https://github.com/upbound/build/) to automate release tagging/branching, building, publishing, and promoting
+
+## Release Schedule
+
+Increasing our release cadence will have multiple benefits for both users and contributors of the project.
+Users will be able to get access to new features and improvements at a faster pace with less wait time.
+Contributors will have more frequent opportunities to include their work into a release,
+which will decrease the stress and pressure we currently feel leading up to each large and long awaited release.
+
+We currently execute on work items in a series of sprints, each lasting 2 weeks.
+Attempting to release at the end of every sprint would be too aggressive and would also dilute the impact of each release.
+We propose a good compromise for release cadence to be releasing a new version at the **end of every 2 sprints**, which translates to **every 4 weeks**, or effectively **monthly**.
+
+## Versioning
+
+While the Crossplane project has already been following [semantic versioning](https://semver.org), we will be very clear in this section about our versioning intent going forward.
+For the reader's convenience, semantic versioning uses a version number format of `MAJOR.MINOR.PATCH`.
+
+In general, before Crossplane is ready for General Availability (GA), we would like to keep the version number below `v1.0`.
+This effectively reserves the first **major** release of Crossplane (`v1.0.0`) for when we are ready to declare it stable and ready for GA.
+
+In the meantime, every monthly release will be a **minor** release, meaning we will be incrementing the minor version number.
+For example, the next 3 monthly releases of Crossplane will be `v0.4.0`, `v0.5.0`, and `v0.6.0`.
+
+### Patch Releases
+
+In addition to monthly minor releases, we will also be able to perform patch releases in the event that a severe issue is identified in the last release.
+Patch releases will increment the `PATCH` version field, for example a patch for the `v0.4.0` release will have a version number of `v0.4.1`.
+
+The exact criteria for what justifies performing a patch release is difficult to identify since every issue is unique.
```

Comment:
How many versions will we commit to supporting.  Will patches be released for `v0.4` once `v0.5.0` is released? What of `v0.3`?